examples/plot_scalable_poly_kernels.py

```diff
@@ -0,0 +1,118 @@
+# is to predict forest cover type from cartographic variables only
+# (no remotely sensed data). After loading, we transform it into a binary
+# classification problem to match the version of the dataset in the
+# LIBSVM webpage [2]_, which was the one used in [1]_.
+
+data = fetch_covtype()
+X, y = data["data"], data["target"]
+
+y[y != 2] = 0
+y[y == 2] = 1  # We will try to separate class 2 from the other 6 classes.
+
+##############################################################################
+# Here we select 20,000 samples for training and 20,000 for testing.
+# To actually reproduce the results in the original Tensor Sketch paper,
+# select 100,000.
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=20_000,
+                                                    test_size=20000,
+                                                    random_state=42)
+
+##############################################################################
+# Now scale features to the range [0, 1] to match the format of the dataset in
+# the LIBSVM webpage, and then normalize to unit length as done in the
+# original Tensor Sketch paper [1].
+
+mm = MinMaxScaler().fit(X_train)
+X_train = normalize(mm.transform(X_train))
+X_test = normalize(mm.transform(X_test))
+
+##############################################################################
+# As a baseline, train a linear SVM on the original features and print the
+# accuracy:
+
+lsvm = LinearSVC()
+lsvm.fit(X_train, y_train)
+lsvm_score = 100*lsvm.score(X_test, y_test)
+print(f"Linear SVM score on raw features: {lsvm_score:.2f} %%")
+
+##############################################################################
+# Then we train linear SVMs on the features generated by
+# :class:`PolynomialSampler` with different values for `n_components`, showing
+# that these kernel feature approximations improve the accuracy of linear
+# classification:
+
+for n_components in [200, 500, 1000, 1500]:
+
+    pipeline = Pipeline(steps=[("kernel_approximator",
+                                PolynomialSampler(
+                                    n_components=n_components,
+                                    degree=4)),
+                               ("linear_classifier", LinearSVC())])
+    pipeline.fit(X_train, y_train)
+    ps_lsvm_score = 100 * pipeline.score(X_test, y_test)
+
+    print(f"Linear SVM score on {n_components} PolynomialSampler " +
+          f"features: {ps_lsvm_score:.2f} %%")
+
+##############################################################################
+# Finally, train a kernelized SVM to see how well :class:`PolynomialSampler`
+# is approximating the performance of the kernel. This of course may take some
+# time, as the SVC class has a relatively poor scalability. This is the
+# reason why kernel approximators are so useful:
+
+from sklearn.svm import SVC
+ksvm = SVC(C=500., kernel="poly", degree=4, coef0=0,
```

Comment:
Also, why do you use `C=500` here, while you use the default in `LinearSVC` ?

Reply:
The reason for that is that the experiment is designed so that if someone changes the number of training samples to `100,000`, we would reproduce the experiment in the original paper as closely as possible. Unfortunately, the paper does not provide the `C`-hyperparameters used, so I tried a few values and `C=1` in the `Linear SVM + PolynomialSampler` and `C=500` for the `Kernel SVM` seem to provide good results, which are also very close to the results reported in the paper. There is some discrepancy due to the stochastic nature of PolynomialSampler and the use of a random training/testing split. Some numbers:

```
Linear SVM with C=0.5 PolynomialSampler features: 79.16 %   -  t= 17.68 s
Linear SVM with C=1 PolynomialSampler features: 79.75 %   -  t= 21.98 s
Linear SVM with C=5 PolynomialSampler features: 80.01 %   -  t= 47.15 s
Linear SVM with C=6 PolynomialSampler features: 80.05 %   -  t= 51.63 s
---------------------------------------------------------------------------
Result in the original paper: 79.34%


Kernel-SVM with C=1 features: 79.91 %%   -  t= 521.89 s 
Kernel-SVM with C=5 features: 81.38 %%   -  t= 496.45 s
Kernel-SVM with C=50 features: 82.85 %%   -  t= 693.35 s
Kernel-SVM with C=100 features: 83.32 %%   -  t= 897.11 s
Kernel-SVM with C=500 features: 84.55 %%   -  t= 2343.52 s
Kernel-SVM with C=700 features: 84.60 %%   -  t= 3007.93 s
---------------------------------------------------------------------------
Result in the original paper: 84.01%
```